lib/src/main/scala/code/chymyst/jc/ReactionSite.scala

```diff
@@ -84,10 +84,14 @@ private[jc] final class ReactionSite(reactions: Seq[Reaction], reactionPool: Poo
 //    moleculesAndValues.foreach{ case (m, v) => m(v) }
 //  }
 
-  private sealed trait ReactionExitStatus
+  private sealed trait ReactionExitStatus {
+    def notFailedWithRetry: Boolean = true
+  }
   private case object ReactionExitSuccess extends ReactionExitStatus
   private case object ReactionExitFailure extends ReactionExitStatus
-  private case object ReactionExitRetryFailure extends ReactionExitStatus
+  private case object ReactionExitRetryFailure extends ReactionExitStatus {
+    override def notFailedWithRetry: Boolean = false
```

Comment:
I agree, it's not technically a double negation, but in the mind "not failed" is "succeeded", so that's a double negation of a sort.

It was just an observation.